### PR TITLE
Add support for adding products to cart by SKU

### PR DESCRIPTION
**Pull request by nikola-mladenovic**

### CODE REVIEW

File: Source/Models.swift [hunk 1]
```diff
@@ -389,8 +389,12 @@ public enum CartUpdateAction: JSONRepresentable {
...
     public var toJSON: [String: Any] {
         switch self {
-        case .addLineItem(let productId, let variantId, let quantity, let supplyChannel, let distributionChannel, let custom):
-            return filterJSON(parameters: ["action": "addLineItem", "productId": productId, "variantId": variantId, "quantity": quantity, "supplyChannel": supplyChannel, "distributionChannel": distributionChannel, "custom": custom])
+        case .addLineItem(let lineItemDraft):
+            return filterJSON(parameters: ["action": "addLineItem", "productId": lineItemDraft.productVariantSelection.values.productId,
+                                           "sku": lineItemDraft.productVariantSelection.values.sku, "quantity": lineItemDraft.quantity,
+                                           "variantId": lineItemDraft.productVariantSelection.values.variantId,
+                                           "supplyChannel": lineItemDraft.supplyChannel, "distributionChannel": lineItemDraft.distributionChannel,
+                                           "externalTaxRate": lineItemDraft.externalTaxRate, "custom": lineItemDraft.custom])
```
Comment (cneijenhuis) [inline Source/Models.swift]: The `externalTaxRate` should not be available via `/me/carts`. It apparently is, but it's a bug I'll get fixed.

File: Source/Models.swift [hunk 2]
```diff
@@ -985,35 +989,59 @@ public struct LineItem: ImmutableMappable {
...
+
+        var values: (productId: String?, variantId: Int?, sku: String?) {
+            switch self {
+                case .productVariant(let productId, let variantId):
+                    return (productId: productId, variantId: variantId, sku: nil)
+                case .sku(let sku):
+                    return (productId: nil, variantId: nil, sku: sku)
+            }
+        }
+    }
```
Comment (cneijenhuis) [inline Source/Models.swift]: 👍

File: Source/Models.swift (hunk 1 above)
Comment (nikola-mladenovic) [inline Source/Models.swift]: I quickly looked up usages of `ExternalTaxRateDraft`, and except for `LineItemDraft` and `CustomLineItemDraft`, it's also being used for setting the `externalTaxRateForShippingMethod` for a `CartDraft`, as well as `setShippingMethod` and `setCustomShippingMethod` update actions.

I guess we should remove it completely, and hopefully this list can help apply the same limits on the platform side.

**Comment by nikola-mladenovic** [comment]
Removed ExternalTaxRateDraft and its usages 👍

File: Source/Models.swift (hunk 1 above)
Comment (nikola-mladenovic) [inline Source/Models.swift]: While removing `ExternalTaxRateDraft`, I also ran into `setCustomShippingMethod` update action which allows for the `taxCategory` to be set, and `changeTaxMode` which allows the tax mode to be changed.

@cneijenhuis  - you can LMK whether we want to address these now (both on the client & platform side), or at some later point in time.

Comment (cneijenhuis) [inline Source/Models.swift]: `setCustomShippingMethod` should not be usable, please write a test.

`changeTaxMode` is special for `/me` and fine.

Comment (nikola-mladenovic) [inline Source/Models.swift]: Removed setCustomShippingMethod update action + added test to make sure it's not usable with raw json method (https://github.com/commercetools/commercetools-ios-sdk/pull/147/commits/da46095bbb9c6963aaee99e7f29999dd4e2fe7fb).

**Pull request merged**